### PR TITLE
Victron node enhancements

### PR DESCRIPTION
**Pull request by pkronstrom**
Includes issues #3, #4, #5, #6 and #13

### CODE REVIEW

File: src/services/victronclient.js [hunk 1]
```diff
@@ -65,13 +70,43 @@ class VictronClient {
...
+     * @param {object} msg a message object received from the dbus-listener
+     */
+    saveToCache(msg) {
+        //let dbusInterface = this.cache[msg.senderName] || {}
+        let dbusInterface = {}
+        if (this.system.cache[msg.senderName]) {
+            dbusInterface = this.system.cache[msg.senderName]
+        } else {
+            // Upon first discovery, request the customname and productname
+            // of the battery. The returned message gets saved to the cache.
```
Comment (tkurki) [inline src/services/victronclient.js:94]: Do we get notified of customName updates somehow?

File: src/services/victronclient.js [hunk 2]
```diff
@@ -16,8 +16,12 @@ class VictronClient {
     constructor(address, port) {
         this.dbusAddress = address || '127.0.0.1'
         this.dbusPort = port || '78'
+
         this.subscriptions = {} // an array of subscription objects [{ "topic": topic, "handler": function }, ...]
+
         this.write
+        this.request
```
Comment (tkurki) [inline src/services/victronclient.js:23]: How about naming these two `this._getValue` and `this._setValue`, underscore to point out they are for _internal use_? This naming obfuscates what these are doing.

Comment (tkurki) [reply src/services/victronclient.js:23]: Is there a possibility that something tries to use these before they are initialised? I guess not, especially when we get to using dbus directly and the connect should never fail.

File: src/services/victronclient.js (hunk 1 above)
Comment (pkronstrom) [reply src/services/victronclient.js:94]: An update to the customName should sends an update message to the dbus, iirc. I need to test this with real hardware when possible, though.

File: src/services/victronclient.js (hunk 2 above)
Comment (pkronstrom) [reply src/services/victronclient.js:23]: Good idea, both are internal, and setValue is exposed via VictronClient.publish() anyways.

Comment (pkronstrom) [reply src/services/victronclient.js:23]: In theory, the relay node could try publishing before the dbus-listener is initialized, good catch. I feel this should be guarded somehow: maybe check that the value of this.connection before writing to dbus.

**Pull request merged**